python/lsst/daf/ingest/index_exposure_task.py

```diff
@@ -0,0 +1,358 @@
+        """
+        parser = pipe_base.ArgumentParser(name=cls._DefaultName)
+        parser.add_argument(
+            '--database', dest='database', required=True,
+            help='SQLite 3 database file name')
+        # Use DatasetArgument to require dataset type be specified on
+        # the command line
+        parser.add_id_argument(
+            '--id', pipe_base.DatasetArgument('dstype'),
+            help='Dataset data id to index')
+        return parser
+
+    def run(self, data_ref, dstype, database):
+        """Index an exposure specified by a data ref and dataset type."""
+        self.index(data_ref.get(dstype), data_ref.dataId, dstype, database)
+
+    def index(self, exp, data_id, dstype, database):
+        """Spatially index an |exposure| or |metadata| object.
+
+        Parameters
+        ----------
+
+        exp : lsst.afw.image.Exposure[DFILU] or lsst.daf.base.PropertySet
+            An afw |exposure| or corresponding |metadata| object.
+
+        data_id : object
+            An object identifying a single exposure (e.g. as used by the
+            butler). It must be possible to pickle `data_id`.
+
+        dstype : str
+            The dataset type of the exposure.
+
+        database : str
+            A file name identifying the SQLite 3 database to populate.
+        """
+        if isinstance(exp, daf_base.PropertySet):
+            md = exp
+            # Map (LTV1, LTV2) to LSST (x0, y0). LSST convention says that
+            # (x0, y0) is the location of the sub-image origin (the bottom-left
+            # corner) relative to the origin of the parent, whereas LTVi encode
+            # the origin of the parent relative to the origin of the subimage.
+            x0 = -md.get('LTV1') if md.exists('LTV1') else 0
+            y0 = -md.get('LTV2') if md.exists('LTV2') else 0
+            width = md.get('NAXIS1')
+            height = md.get('NAXIS2')
+            wcs = afw_image.makeWcs(md, False)
+        else:
+            x0 = exp.getX0()
+            y0 = exp.getY0()
+            width = exp.getWidth()
+            height = exp.getHeight()
+            wcs = exp.getWcs()
+        # Create a convex polygon containing the exposure pixels. When sphgeom
+        # gains support for non-convex polygons, this could be changed to map
+        # exposure.getPolygon() to a spherical equivalent, or to subdivide box
+        # edges in pixel space to account for non linear projections. This
+        # would have higher accuracy than the current approach of connecting
+        # corner sky coordinates with great circles.
+        corners = [
+            wcs.pixelToSky(x0 - 0.5, y0 - 0.5).toIcrs(),
+            wcs.pixelToSky(x0 - 0.5, y0 + height - 0.5).toIcrs(),
+            wcs.pixelToSky(x0 + width - 0.5, y0 + height - 0.5).toIcrs(),
+            wcs.pixelToSky(x0 + width - 0.5, y0 - 0.5).toIcrs()
+        ]
+        corners = [
+            LonLat.fromRadians(c.getLongitude().asRadians(),
```

Comment:
My understanding is that sphgeom doesn't do any conversions between different types of celestial coordinate systems (e.g. galactic, ecliptic), but we don't actually need to afw.coord to do any of that either, as we actually always use ICRS.  There's a proposal coming from the AstroPy integration side to drop afw.coord in favor of AstroPy coordinates in Python, and just have a spherical point class (which could be something from sphgeom) in C++.